flink-runtime/src/main/java/org/apache/flink/runtime/minicluster/MiniClusterJobClient.java

```diff
@@ -106,7 +108,25 @@ public JobID getJobID() {
 
     @Override
     public CompletableFuture<Map<String, Object>> getAccumulators() {
-        return getJobExecutionResult().thenApply(JobExecutionResult::getAllAccumulatorResults);
+        //TODO: this is not the only case when we need to get the accumulators from JobExecutionResult.
+        // It is needed also for UnalignedCheckpointCompatibilityITCase savepoints tests to pass.
+        // What is the complete condition ?
+        if (!miniCluster.isRunning()) {
```

Comment:
I'm wondering if the `miniCluster.isRunning()` is not a proper condition ? Since a mini-cluster could be viewed as a cluster that could run multiple jobs, the miniCluster would still be able to run after the job is done logically ? Perhaps we could use `jobResultFuture.isDone()`

Reply:
agree `miniCluster.isRunning()` is more a side effect of `PerJobMiniClusterFactoryTest.testJobExecution` stopping the minicluster when the job is done.  `jobResultFuture.isDone()` is definitely a better condition.